core/domain/exp_domain.py

```diff
@@ -882,40 +887,23 @@ def validate(self, interaction, exp_param_specs_dict):
             raise utils.ValidationError(
                 'Expected answer group rules to be a list, received %s'
                 % self.rule_specs)
-        if len(self.rule_specs) < 1:
+
+        if len(self.rule_specs) < 1 and self.training_data == []:
```

Comment:
len(self.training_data) < 1

(or have them both be == 0, but let's do it the same way for both lists)

Reply:
Done